### ISSUE TITLE
Session data (authenticated object) is empty after page refresh

### ISSUE THREAD

**Issue by ashwini709** [simplabs/ember-simple-auth#1405]
Using **ember-simple-auth version 1.4.0**

Following are my settings for `ember-simple-auth`
```
  ENV['ember-simple-auth'] = {
    authenticator: 'authenticator:oauth2',
    authorizer: 'authorizer:oauth2',
    routeAfterAuthentication: 'abc'
  };
```

Refreshing page after login, takes me back to login page.
My localStorage is still intact: 

```
ember_simple_auth-session: {"authenticated":{"authenticator":"authenticator:oauth2","access_token":"....","token_type":"bearer","expires_in":86400,"refresh_token":"...","created_at":1500616620,"expires_at":1500703020737}}

ember_simple_auth:session: {"secure":{"authenticator":"authenticator:oauth2","access_token":"...","token_type":"bearer","expires_in":86400,"refresh_token":"...","created_at":1500601942,"expires_at":1500688342318}}
```

Using custom version of `oauth2-password-grant`.

I realized the `session data persists or restore method for local-storage` is never called after page refresh. `restore for local-storage` will only be called after `window.addEventListener('storage', bind(this, this._handleStorageEvent));` and since its never changing the session `authenticated` object is empty.

@marcoow Is this a bug? Should it try to restore local storage data in every init call?

**Comment by marcoow** [comment]
I could not reproduce this in the dummy app. There also are a few problems in your code apparently which might be causing this:

* You cannot configure the authenticator and authorizer in `config/environment.js` anymore. Have a look at the README for more info [here](https://github.com/simplabs/ember-simple-auth#authenticators) and [here](https://github.com/simplabs/ember-simple-auth#authorizers).
* `ember_simple_auth:session: {"secure":{"authenticator":"authenticator:oauth2","access_token":"...","token_type":"bearer","expires_in":86400,"refresh_token":"...","created_at":1500601942,"expires_at":1500688342318}}` is the old data format in `localStorage` and not used anymore. I'm also not sure which of the 2 values you posted for the `ember_simple_auth:session` key you actually have in your `localStorage`.

**Comment by ashwini709** [comment]
Thank  you for the reply. I corrected the config you mentioned and cleaned the localstorage, yes `ember_simple_auth:session` was the old one. It was still not working and I realized why the session storage does not restore on refresh, it refreshes on `applicationroute's beforeModel` hook as per [this](https://github.com/simplabs/ember-simple-auth/blob/master/addon/instance-initializers/setup-session-restoration.js#L12) and the `beforeModel` is triggered by `runBeforeModelHook`, which works only after ember version > 2.
My ember version is `1.13.13`, Will work fine after ember version upgrade I think.
